core/src/main/java/io/parsingdata/metal/data/selection/ByToken.java

```diff
@@ -78,5 +78,22 @@ private static ParseValueList getAllValuesRecursive(final ParseGraph graph, fina
         if (head.isGraph()) { return tailResults.add(getAllValuesRecursive(head.asGraph(), definition)); }
         return tailResults;
     }
+    
+    public static ParseItemList getAllRoots(final ParseGraph graph, final Token definition) {
+        checkNotNull(graph, "graph");
+        checkNotNull(definition, "definition");
+        return (graph.getDefinition() == definition && !graph.isEmpty() ? ParseItemList.create(graph) : ParseItemList.EMPTY)
+            .add(getAllRootsRecursive(graph.head, graph, definition))
+            .add(getAllRootsRecursive(graph.tail, graph, definition));
```

Comment:
I like this alignment of the two `.add`s, you could do that below as well, although that might look a bit too weird?